govcd/catalog.go

```diff
@@ -62,17 +62,17 @@ func NewAdminCatalog(client *Client) *AdminCatalog {
 // Deletes the Catalog, returning an error if the vCD call fails.
 // Link to API call: https://code.vmware.com/apis/220/vcloud#/doc/doc/operations/DELETE-Catalog.html
 func (catalog *Catalog) Delete(force, recursive bool) error {
-	adminCatalogHREF, err := url.ParseRequestURI(catalog.Catalog.HREF)
-	if err != nil {
-		return fmt.Errorf("error parsing admin catalog's href: %v", err)
-	}
+
+	adminCatalogHREF := catalog.client.VCDHREF
+	// slice only numeric part from ID:"urn:vcloud:catalog:97384890-180c-4563-b9b7-0dc50a2430b0"
+	adminCatalogHREF.Path += "/admin/catalog/" + catalog.Catalog.ID[19:]
```

Comment:
You have replaced the "[19:]" with some function in one of the previous PRs. Would be nice to have it here as well.

Reply:
I am reverting back previous implementation as it wasn't good enough, it worked only for admin user.